src/contexts/UserContext.tsx

```diff
@@ -0,0 +1,41 @@
+import { createContext, ReactNode, useEffect, useMemo, useState } from 'react'
+import { User } from '@api/interfaces'
+import { useFetchCurrentUser } from '@api/hooks'
+
+interface UserContextProps {
+  user?: User
+  reloadUser: () => void
+  isLoading: boolean
+}
+
+export const UserContext = createContext<UserContextProps>(
+  {} as UserContextProps
+)
+
+interface UserProviderProps {
+  children: ReactNode
+}
+
+export const UserProvider = ({ children }: UserProviderProps) => {
+  const { data, isLoading, refetch, isError } = useFetchCurrentUser()
+
+  const [user, setUser] = useState<User | undefined>(undefined)
+
+  useEffect(() => {
+    if (isError) return setUser(undefined)
```

Comment:
Same as above.

Reply:
The data type returned from useQuery is User | undefined

Reply:
The type of data from useQuery is User | undefined